src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+        update contactsAddrs.addrs;
+        Test.stopTest();
+    
+        // verify that the Contact and Contacts don't share the same address and it's new for the Contact!
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c 
+        from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assertEquals(null, contact.MailingStreet); //address information has been cleared because address is no longer the default
+            system.assertEquals(null, contact.MailingCity); //address information has been cleared because address is no longer the default
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertEquals(null, contact.Current_Address__c); //Contact no longer has a current address, since address is no longer the defaults
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Set Contact address override.  
+    verify: Contact's address matches override
+    **********************************************************************************************************/            
+    @isTest
+    public static void setAddrOverride() {
+        configSettings();
+        
+        // this creates a default Address for the Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // create new non-default address
+        Address__c addrNew = new Address__c();
+        addrNew.MailingStreet__c = 'street override';
+        addrNew.MailingCity__c = 'city override';
+        addrNew.Default_Address__c = false;
+        // assign it to the second Contact
+        addrNew.Parent_Contact__c = contactsAddrs.contacts[1].Id;
+        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrNew;
+        
+        // set Contact's address override
+        contactsAddrs.contacts[1].Current_Address__c = addrNew.Id;
+        contactsAddrs.contacts[1].is_Address_Override__c = true;
+        
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts[0];
+    
+        // now let's update the default Address for both Contacts
+        contactsAddrs.addrs[0].MailingStreet__c = 'New Street';
+        contactsAddrs.addrs[0].MailingCity__c = 'New City';
+        contactsAddrs.addrs[1].MailingStreet__c = 'New Street';
+        contactsAddrs.addrs[1].MailingCity__c = 'New City';
+
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.addrs;
+        Test.stopTest();
+        
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.